Target-MVD-Mathias/Views/Modal.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  Modal.swift
+//  Target-MVD-Mathias
+//
+//  Created by Mathias Cabano on 10/6/17.
+//  Copyright © 2017 TopTier labs. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+protocol Modal {
+  var backgroundView: UIView! { get }
+  var dialogView: UIView! { get set }
+  
+  func show(animated: Bool)
+  func dismiss(animated: Bool)
+}
+
+extension Modal where Self: UIView {
+  
+  func show(animated: Bool) {
+    self.backgroundView.alpha = 0
+    UIApplication.shared.delegate?.window??.rootViewController?.view.addSubview(self)
```

Comment:
What is this two `??` followed by a `.` means here? 
Just for curiosity, first time i see this

Reply:
It's because window's type is UIWindow??, I don't know why

Reply:
Weird.. Is the window var of your AppDelegate of UIWindow?? type?

Reply:
Maybe what you are looking for is: `UIApplication.shared.keyWindow`

Reply:
No, the window var of the App delegate is of type UIWindow?. I'll try the keyWindow option